tests/test_config.py

```diff
@@ -46,6 +47,13 @@ def test_config_from_json():
     common_object_test(app)
 
 
+def test_config_from_file():
```

Comment:
Make this an optional dependency with [`pytest.importorskip`](https://docs.pytest.org/en/latest/skipping.html#skipping-on-a-missing-import-dependency) so users can run `pytest` directly for most contributions.